File: lab1.c
#include <math.h>
#include <stdio.h>     

int main()
{
	float a, x, s, f, g, y;
	int funct;
	double pi = 3.141592653589;
	printf("Введите x: ");
	scanf_s("%f", &x);
	printf("Введите a: ");
	scanf_s("%f", &a);
	printf("Выберите какую ф-ию посчитать: f - 1, g -2, y - 3 ");
	scanf_s("%i", &funct);

	if (funct == 1) {
		f = (sin(pi * (10 * a * a + 37 * a * x + 7 * x * x)));
		printf("%f", f);
	}
	else if (funct == 2) {
		g = -1 * (2 * (-5 * a * a + 3 * a * x + 2 * x * x) / (5 * a * a + 9 * a * x - 2 * x * x));
		printf("%f", g);
	}
	else if (funct == 3) {
		y = (log(-5 * a * a - 16 * a * x + 16 * x * x + 1) / log(2));
		printf("%f", y);
	}
	else {
		printf("ERROR");
	}
}

File: lab1.py
from math import *

a = float(input())
x = float(input())

g = -1 * (2 * (-5 * a**2 + 3 * a * x + 2 * x**2) / (5 * a**2 + 9 * a * x - 2 * x**2))

f = (sin(pi * (10 * a**2 + 37 * a * x + 7 * x**2)))

y = (log(-5 * a**2 - 16 * a * x + 16 * x**2 + 1) / log(2))

print(g)
print(f)
print(y)